langchain/agents/openai_functions_multi_agent/base.py

```diff
@@ -138,7 +138,8 @@ def _parse_ai_message(message: BaseMessage) -> Union[List[AgentAction], AgentFin
                 message_log=[message],
             )
             final_tools.append(_tool)
-        return final_tools
+        if len(final_tools) > 0:
```

Comment:
when would this happen?

Reply:
The AI occasionally calls the function that does not have any actions(ie. performing `tool_selection` with the argument `actions` being an empty array.) 

The agent is then prompted again(because not reaching the line AgentFinish)

Couple this problem with a low-temperature setting in the model, this gets stuck in a loop, especially when the agent is prompted to use functions more often.